lib/cuda_gauge_field.cpp

```diff
@@ -141,14 +143,15 @@ namespace quda {
       }
 
       extractGaugeGhost(*this, send_d, true, link_dir*nDim); // get the links into contiguous buffers
+      qudaDeviceSynchronize(); // synchronize before issuing mem copies in different streams - could replace with event post and wait
```

Comment:
worth backporting to 1.1?

Reply:
The reason this extra sync was added, was because previously we were relying on null-stream semantics, and so the subsequent copies could not start until the extract kernel (issued to the NULL stream) was complete.  With GK, we move to named streams, hence the need for this synchronization.